semantic_conventions/http-common.yaml

```diff
@@ -5,10 +5,51 @@ groups:
     prefix: http
     attributes:
       - id: request.method
-        type: string
+        type:
+          allow_custom_values: true
+          members:
+            - id: connect
+              value: "CONNECT"
+              brief: 'CONNECT method.'
+            - id: delete
+              value: "DELETE"
+              brief: 'DELETE method.'
+            - id: get
+              value: "GET"
+              brief: 'GET method.'
+            - id: head
+              value: "HEAD"
+              brief: 'HEAD method.'
+            - id: options
+              value: "OPTIONS"
+              brief: 'OPTIONS method.'
+            - id: patch
+              value: "PATCH"
+              brief: 'PATCH method.'
+            - id: post
+              value: "POST"
+              brief: 'POST method.'
+            - id: put
+              value: "PUT"
+              brief: 'PUT method.'
+            - id: trace
+              value: "TRACE"
+              brief: 'TRACE method.'
+            - id: other
+              value: "other"
+              brief: 'Any custom HTTP method that the instrumentation has no prior knowledge of.'
         requirement_level: required
         brief: 'HTTP request method.'
         examples: ["GET", "POST", "HEAD"]
+        note: |
+          HTTP request method SHOULD be one of the methods defined in [RFC9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-methods)
+          or the PATCH method defined in [RFC5789](https://www.rfc-editor.org/rfc/rfc5789.html).
+          Instrumentation MAY additionally support the closed set of custom HTTP methods defined in
```

Comment:
IMHO, implementations MUST additionally provide a way to allow other methods.

Reply:
there is no reason to require it for specialized instrumentations (e.g. HTTP web framework that does not allow customization, WebDAV clients, or other opinionated non-otel-authored instrumentations).

For OTel-owned instrumentations SHOULD is enough.

Reply:
@Oberon00 , can you common on the MUST here considering my comment ⬆️

Reply:
@lmolkova @Oberon00 does this address both your concerns? https://github.com/open-telemetry/semantic-conventions/pull/17#discussion_r1230382772

> If the HTTP instrumentation could end up converting valid HTTP request methods to OTHER, then it MUST provide a way to override the list of known HTTP methods

Reply:
If we start distinguishing specialized vs general-purpose here, we need to define what it is. But it sounds OK.

Note that it _could_ still make sense though, e.g. if you want to monitor a particular HTTP method that you get requests with despite not supporting it.

Reply:
I believe this is addressed with @trask's suggestion